### PR TITLE
Get weets

### PR DESCRIPTION
**Pull request by CristianRWolox**
## Summary

- Make request to the `weets` endpoint.
- Validate pagination.
- Receive page and limit through query string.

## Trello Card

[Get Weets](https://trello.com/c/ivyqt9UP/36-obtener-weets)

### CODE REVIEW

File: app/middlewares/param-validation.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+const { conflictError } = require('../errors');
+
+const paramValidation = param => (req, res, next) => {
+  const queryParam = req.query[param];
+  if (!queryParam) {
+    return next(conflictError('param schema', `The ${param} parameter is required.`));
+  }
+
+  return next();
+};
```
Comment (dbenavidesv) [inline app/middlewares/param-validation.js:10]: You can check this with the schema validator

File: app/services/weets.js [hunk 2]
```diff
@@ -12,4 +13,15 @@ const getWeet = async () => {
   }
 };
 
-module.exports = getWeet;
+const findAndCountAll = async (user_id, page, limit) => {
+  const data = await Weet.findAndCountAll({
+    where: {
+      user_id
+    },
+    offset: page * limit,
```
Comment (dbenavidesv) [inline app/services/weets.js:21]: I think this should be (page * limit - limit)

File: app/services/weets.js [hunk 3]
```diff
@@ -12,4 +13,15 @@ const getWeet = async () => {
   }
 };
 
-module.exports = getWeet;
+const findAndCountAll = async (user_id, page, limit) => {
```
Comment (dbenavidesv) [inline app/services/weets.js:16]: variables should be in camelCase

File: app/middlewares/param-validation.js (hunk 1 above)
Comment (CristianRWolox) [reply app/middlewares/param-validation.js:10]: Done

File: app/controllers/weet/get-weets.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+const logger = require('../../logger');
+const { findAndCountAll } = require('../../services/weets');
+const { pagination } = require('../../utils/pagination');
+
+exports.getWeets = async (req, res, next) => {
+  const { user_id } = req.query;
```
Comment (dbenavidesv) [inline app/controllers/weet/get-weets.js:6]:
```suggestion
  const { user_id: userId } = req.query;
```

File: app/controllers/weet/get-weets.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  const { page, limit } = pagination(req.query.page, req.query.limit);
+  try {
+    const weets = await findAndCountAll(user_id, page, limit);
```
Comment (dbenavidesv) [inline app/controllers/weet/get-weets.js:9]:
```suggestion
    const weets = await findAndCountAll(userId, page, limit);
```

**Pull request merged**